File: String/AnagramPalindrome.py
# Anagram Palindrome
# Given a string S, Check if characters of the given string 
# can be rearranged to form a palindrome. 

from collections import Counter

def check(s:str)->bool:
    h = Counter(s)
    c = 0
    for (key,val) in h.items():
        if(val==1):
            c+=1
    res = "YES" if(c==1) else "NO"
    return res


if __name__=='__main__':
    s1 = "geeksogeeks"
    s2  = "geeksforgeeks"
    print(check(s1),end="\n") # YES
    print(check(s2),end="\n") # NO

File: String/AnagramPalindrome.cxx
// Anagram Palindrome
// Given a string S, Check if characters of the given string 
// can be rearranged to form a palindrome. 

#include <bits/stdc++.h>
using namespace std;

class Solution{
    public:
    static void check(string s){
        unordered_map<char,int> h;
        auto c = 0;
        for(auto x : s)
            h[x]++;
        for(auto y : h){
            if(y.second==1)
                c++;
        }
        (c==1) ? printf("%s\n", "YES") : printf("%s\n", "NO");
    }
};

int main(){
    string s1 = "geeksogeeks";
    string s2 = "geeksforgeeks";
    Solution::check(s1); // YES
    Solution::check(s2); // NO
    return 0;
}